lib/smb2/file.rb

```diff
@@ -58,23 +79,23 @@ def inspect
   # @param length [Fixnum,nil] number of bytes to read. If this is nil, read
   #   the whole file, i.e. {#size} bytes.
   # @return [String] full contents of the remote file
-  def read(offset: 0, length: nil)
+  def read(length = nil, offset: self.pos)
```

Comment:
Yes. It's a little weird though, since [IO#read](http://ruby-doc.org/core-2.2.2/IO.html#method-i-read) just uses the current position pointer; it doesn't take an offset at all. I thought about just changing this to mirror that exactly, but it's a weird API because of the bizarre and very un-Ruby-like `outbuf` argument. What do you think?